src/AppBundle/Membership/MembershipRequest.php

```diff
@@ -0,0 +1,154 @@
+ */
+class MembershipRequest
+{
+    /**
+     * @AssertGender
+     */
+    public $gender;
+
+    /**
+     * @Assert\NotBlank(message="common.first_name.not_blank")
+     * @Assert\Length(
+     *   min=2,
+     *   max=50,
+     *   minMessage="common.first_name.min_length",
+     *   maxMessage="common.first_name.max_length"
+     * )
+     */
+    public $firstName;
+
+    /**
+     * @Assert\NotBlank(message="common.first_name.not_blank")
+     * @Assert\Length(
+     *   min=2,
+     *   max=50,
+     *   minMessage="common.last_name.min_length",
+     *   maxMessage="common.last_name.max_length"
+     * )
+     */
+    public $lastName;
+
+    /**
+     * @Assert\NotBlank(message="common.email.not_blank")
+     * @Assert\Email(message="common.email.invalid")
+     */
+    public $emailAddress;
+
+    /**
+     * @AssertPhoneNumber(defaultRegion="FR")
+     */
+    private $phone;
+
+    /**
+     * @Assert\NotBlank(message="adherent.birthdate.not_blank")
+     * @Assert\Range(max="-15 years", maxMessage="adherent.birthdate.minimum_required_age")
+     */
+    private $birthdate;
+
+    /**
+     * @Assert\Length(max=150, maxMessage="common.address.max_length")
+     */
+    public $address;
+
+    /**
+     * @AssertFrenchPostalCode(message="common.postal_code.invalid")
+     */
+    public $postalCode;
+
+    /**
+     * @AssertFrenchCity(message="common.city.invalid")
+     */
+    public $city;
+
+    /**
+     * @AssertUnitedNationsCountry(message="common.country.invalid")
+     */
+    public $country;
+
+    /**
+     * @AssertActivityPosition
+     */
+    public $position;
+
+    /**
+     * @Assert\NotBlank(groups="Registration")
+     * @Assert\Length(min=8, minMessage="adherent.plain_password.min_length")
+     */
+    public $password;
+
+    /**
+     * @Assert\IsTrue(message="common.conditions.not_accepted")
+     */
+    public $conditions;
+
+    public function __construct(AdherentRepository $repository)
+    {
+        $this->repository = $repository;
+        $this->country = 'FR';
+        $this->gender = Genders::MALE;
+        $this->position = ActivityPositions::EMPLOYED;
+        $this->conditions = false;
+    }
+
+    public static function createFromDoctrine(ManagerRegistry $manager)
+    {
+        return new self($manager->getRepository(Adherent::class));
+    }
+
+    public function setPhone(PhoneNumber $phone = null)
+    {
+        $this->phone = $phone;
+    }
+
+    public function getPhone()
+    {
+        return $this->phone;
+    }
+
+    public function setBirthdate(\DateTime $birthdate)
+    {
+        $this->birthdate = $birthdate;
+    }
+
+    public function getBirthdate()
+    {
+        return $this->birthdate;
+    }
+
+    /**
+     * @Assert\Callback
+     */
+    public function checkEmailAddressUniquess(ExecutionContextInterface $context)
+    {
+        $adherent = $this->repository->findByEmail($this->emailAddress);
```

Comment:
I would personnally prefer an external service to check the DTO uniqueness, would it be possible?